### DISCUSSION TITLE
Unwanted log "HEADERS ALREADY SENT:"

### DISCUSSION THREAD

**Discussion by henrykbrzoska** [OpenMage/magento-lts#1353] [Questions & Answers] [answered]
https://github.com/OpenMage/magento-lts/blob/1.9.4.x/app/code/core/Mage/Core/Controller/Response/Http.php:52
dev should **not** be prevented from sending a response without leaving footprint.
After this canSendHeaders() is another one from Zend which throw exception.

This is problematic, when system is strongly integrated.
This should be commented out.

**Comment by Flyingmana** [comment]
Thats a PHP / HTTP thing. As soon as a Developer is sending anything out, you can nott send any headers anymore to the browser.
The reason is, Browsers interpret what you send according to the Headers which were send before, and sending headers later will be ignored by Browsers. Thats why this is logged and prevented, because from this point on the behavior of the application is likely already broken.

**Reply by henrykbrzoska** [reply]
100% right!
Because of that we have(later in magento logic)
lib/Zend/Controller/Response/Abstract.php:338 with special 'if' and additional logic for this, and it works(but not log everytime it happen without any logic).

so
we shoud comment out it, or add same logic.

example where can be useful - communicate with outside solution, after add something to cart. You can send json to client after save quote, and do curl outside after this if it not more important than UX/speed.

**Reply by colinmollenhour** [reply, marked as answer]
I think you're looking for something related to this: https://github.com/OpenMage/magento-lts/pull/113

**Reply by henrykbrzoska** [reply]
> I think you're looking for something related to this: #113

yes. I did what i need, but there is still double checked canSendHeaders(), first with unnecessary Mage::log
IMO it shouldn't be there.
